File: src/go/02_readfile/main.go
package main

import (
	"fmt"
	"io/ioutil"
	"os"
	"strings"
)

func main() {
	args := os.Args

	if len(args) < 2 {
		fmt.Println("need 1 arguments.")
		os.Exit(1)
	}

	// ファイル読み込み
	infile := args[1]
	b, err := ioutil.ReadFile(infile)
	if err != nil {
		panic(err)
	}
	// byte型なんでstringに変換
	s := string(b)

	// 改行文字で区切って配列に
	lines := strings.Split(s, "\n")
	i := 0
	for _, line := range lines {
		line = strings.TrimSpace(line)
		if line == "" {
			continue
		}
		i++
		line = fmt.Sprintf("%03d:%s", i, line)
		fmt.Println(line)
	}
}

File: src/julia/02_readfile/main.jl
#!/usr/local/bin/julia

using Printf

if length(ARGS) < 1
  println("need 2 arguments.")
  exit(1)
end

open(ARGS[1], "r") do fp
  i = 0
  for line in eachline(fp)
    i = i + 1
    @printf "%03d:%s\n" i line
  end
end
